tests/attestation_plugin_cert/enc/enc.cpp

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <mbedtls/ctr_drbg.h>
+#include <mbedtls/entropy.h>
+#include <mbedtls/pk.h>
+#include <mbedtls/rsa.h>
+#include <openenclave/attestation/attester.h>
+#include <openenclave/attestation/sgx/evidence.h>
+#include <openenclave/attestation/verifier.h>
+#include <openenclave/bits/evidence.h>
+#include <openenclave/edger8r/enclave.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/report.h>
+#include <openenclave/internal/sgx/plugin.h>
+#include <openenclave/internal/tests.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "tls_t.h"
+
+// This is the claims validation callback. A TLS connecting party (client or
+// server) can verify the passed in claims to decide whether to
+// accept a connection request
+oe_result_t enclave_claims_verifier(
+    oe_claim_t* claims,
+    size_t claims_length,
+    void* arg)
+{
+    oe_result_t result = OE_VERIFY_FAILED;
+
+    (void)arg;
+    OE_TRACE_INFO("enclave_claims_verifier is called with claims:\n");
+
+    for (size_t i = 0; i < claims_length; i++)
+    {
+        oe_claim_t* claim = &claims[i];
+        if (strcmp(claim->name, OE_CLAIM_SECURITY_VERSION) == 0)
+        {
+            uint32_t security_version = *(uint32_t*)(claim->value);
+            // Check the enclave's security version
+            if (security_version < 1)
+            {
+                OE_TRACE_ERROR(
+                    "identity->security_version checking failed (%d)\n",
+                    security_version);
+                goto done;
+            }
+        }
+        // Dump an enclave's unique ID, signer ID and Product ID. They are
+        // MRENCLAVE, MRSIGNER and ISVPRODID for SGX enclaves. In a real
```

Comment:
In my opinion, SGX specific info should only appear in a file where "sgx" is in the filename or path to the file.  Much of the rest of this file is agnostic.  I'd recommend moving this function (and any other sgx-only code in this file) to a separate file with sgx in the filename.

Reply:
Renamed the function name to `sgx_enclave_claims_verifier`.